code/modules/reagents/machinery/chem_master.dm

```diff
@@ -433,15 +441,26 @@ var/global/list/pillIcon2Name = list("oblong purple-pink", "oblong green-white",
 	var/dat = list()
 	dat += "<HR>"
 	dat += "<div class='pillIconsContainer'>"
-	for(var/i = 1 to MAX_PILL_SPRITE)
+	for(var/i = 1 to MAX_PILL_SPRITE/2) // show the first half, rounding needed for odd
 		dat += {"<a href="?src=\ref[src]&pill_sprite=[i]" class="pillIconWrapper[i == text2num(pillsprite) ? " linkOnMinimal" : ""]">
 					<div class="pillIcon">
 						[pill_icon_cache[i]]
 					</div>
 				</a>"}
 		if (i%10 == 0)
-			dat +="<br>"
-	dat += "</div>"
+			dat += "<br>"
+
+	if (iconToggle)
+		for(var/i = (MAX_PILL_SPRITE/2)+1 to MAX_PILL_SPRITE) // show the second half, rounding needed for odd
```

Comment:
yikes
The functionality is good but this is really bad copypaste
Try adding a new argument to `generate_pill_icon_div()`, then modify the line inside that proc where it says `for(var/i = 1 to MAX_PILL_SPRITE)` and replace `MAX_PILL_SPRITE` with your new argument
Your final proc should look something like:
`/obj/machinery/chem_master/proc/generate_pill_icon_div(var/the_number_up_to_which_we_want_to_draw_pill_icons)
	var/dat = list()
	dat += "<HR>"
	dat += "<div class='pillIconsContainer'>"
	for(var/i = 1 to the_number_up_to_which_we_want_to_draw_pill_icons)
		dat += {"<a href="?src=\ref[src]&pill_sprite=[i]" class="pillIconWrapper[i == text2num(pillsprite) ? " linkOnMinimal" : ""]">
					<div class="pillIcon">
						[pill_icon_cache[i]]
					</div>
				</a>"}
		if (i%10 == 0)
			dat +="<br>"
	dat += "</div>"
	return dat`
Then when you call `generate_pill_icon_div()` you can give it an argument of 20 or 40 depending on what the setting is set to

Reply:
already working on it